plugins/pulp_rpm/plugins/importers/yum/depsolve.py

```diff
@@ -421,8 +432,6 @@ def get_requirements(self, units):
         :rtype:     generator
         """
         for segment in paginate(units):
-            unit_ids = [unit.id for unit in segment]
-            fields = ['requires', 'id']
-            for result in models.RPM.objects.filter(id__in=unit_ids).only(*fields):
-                for require in result.requires or []:
+            for unit in segment:
+                for require in unit.requires or []:
```

Comment:
The dockblock is wrong, but supports my suspicions mentioned above (I didn't even read it, woops). This function is now being passed instances of the RPM unit type.

I've come up with a couple other good reasons to run the instances back through the unit model manager, so I think I'll revert this, and add a comment outlining the various reasons why it's not totally weird do do this.